ydb/library/yql/public/purecalc/io_specs/arrow/ut/test_spec.cpp

```diff
@@ -0,0 +1,224 @@
+#include <library/cpp/testing/unittest/registar.h>
+
+#include <ydb/library/yql/public/purecalc/common/interface.h>
+#include <ydb/library/yql/public/purecalc/io_specs/arrow/spec.h>
+#include <ydb/library/yql/public/purecalc/ut/lib/helpers.h>
+
+#include <ydb/library/yql/public/udf/arrow/udf_arrow_helpers.h>
+#include <arrow/array/builder_primitive.h>
+
+namespace {
+
+template <typename T>
+struct TVectorStream: public NYql::NPureCalc::IStream<T*> {
+    TVector<T> Data_;
+    size_t Index_ = 0;
+
+public:
+    TVectorStream(TVector<T> items)
+        : Data_(std::move(items))
+    {
+    }
+
+    T* Fetch() override {
+        return Index_ < Data_.size() ? &Data_[Index_++] : nullptr;
+    }
+};
+
+
+template <typename T>
+class TCallbackConsumer: public NYql::NPureCalc::IConsumer<T*> {
+private:
+    using TCallable = void (*)(const T*);
+    int I_ = 0;
+    TCallable Callback_;
+
+public:
+    TCallbackConsumer(TCallable callback)
+        : Callback_(std::move(callback))
+    {
+    }
+
+    void OnObject(T* t) override {
+        I_ += 1;
+        Callback_(t);
+    }
+
+    void OnFinish() override {
+        UNIT_ASSERT(I_ > 0);
+    }
+};
+
+
+using ExecBatchStreamImpl = TVectorStream<arrow::compute::ExecBatch>;
+using ExecBatchConsumerImpl = TCallbackConsumer<arrow::compute::ExecBatch>;
+
+
+static constexpr i64 value = 19;
+static constexpr ui64 bsize = 9;
+
+arrow::compute::ExecBatch MakeBatch() {
+    TVector<ui64> data(bsize);
+    TVector<bool> valid(bsize);
+    std::iota(data.begin(), data.end(), 1);
+    std::fill(valid.begin(), valid.end(), true);
+
+    arrow::UInt64Builder builder;
+    ARROW_OK(builder.Reserve(bsize));
+    ARROW_OK(builder.AppendValues(data, valid));
+
+    arrow::Datum array(ARROW_RESULT(builder.Finish()));
+    arrow::Datum scalar(std::make_shared<arrow::Int64Scalar>(value));
+
+    TVector<arrow::Datum> batchArgs = {array, scalar};
+    return arrow::compute::ExecBatch(std::move(batchArgs), bsize);
+}
+
+void AssertBatch(const arrow::compute::ExecBatch* batch) {
+    arrow::Datum first = batch->values[0];
+    arrow::Datum second = batch->values[1];
+
+    UNIT_ASSERT(first.is_array());
+    const auto& array = *first.array();
+    UNIT_ASSERT_EQUAL(array.length, bsize);
```

Comment:
UNIT_ASSERT_VALUES_EQUAL

Reply:
Yeah, this is a nice nit, but I've reimplemented testing assertions, so it's not relevant anymore for this PR (but still very valuable for the future patchsets).